starport_template/lib/pages/assets_portfolio_page.dart

```diff
@@ -79,16 +80,27 @@ class _AssetsPortfolioPageState extends State<AssetsPortfolioPage> {
     );
   }
 
-  Padding _buildGradientAvatar(BuildContext context) {
+  Widget _gradientAvatar(BuildContext context) {
     return Padding(
       padding: EdgeInsets.all(CosmosTheme.of(context).spacingL),
       child: Align(
         alignment: Alignment.centerLeft,
-        child: SizedBox(height: 35, child: GradientAvatar(stringKey: selectedWallet.publicAddress)),
+        child: SizedBox(
+          height: 35,
+          child: InkWell(
+            onTap: () => _onTapAvatar(context),
+            child: GradientAvatar(stringKey: selectedWallet.publicAddress),
+          ),
+        ),
       ),
     );
   }
 
+  Future<void> _onTapAvatar(BuildContext context) async {
+    await Navigator.of(context).push(MaterialPageRoute(builder: (context) => const TransactionHistoryPage()));
+    StarportApp.walletsStore.getBalances(selectedWallet.publicAddress);
```

Comment:
do we need to fetch balances after the `TransactionHistoryPage` is closed?

Reply:
The `selectedWallet` might be changed by the user in this history page and it needs to reflect the latest balances from the current selected wallet

Reply:
we should make it a part of changing the selected wallet then, so that each change will trigger this reload, instead of doing it in UI. you can make it inside the setter for `selectedWalletIndex` probably